### PR TITLE
Replace ARCH sed macros with Go templates

### PR DESCRIPTION
**Pull request by medyagh**

### CODE REVIEW

**Comment by tstromberg** [comment]
@minikube-bot OK to test

File: pkg/minikube/assets/addons.go [hunk 1]
```diff
@@ -382,8 +383,10 @@ func addMinikubeDirToAssets(basedir, vmpath string, assets *[]CopyableFile) erro
 func GenerateTemplateData(cfg config.KubernetesConfig) interface{} {
 	opts := struct {
 		ImageRepository string
+		ArchTag         string
 	}{
 		ImageRepository: cfg.ImageRepository,
+		ArchTag:         runtime.GOARCH,
```
Comment (tstromberg) [inline pkg/minikube/assets/addons.go]: How about just "Arch": it's shorter.

File: deploy/addons/storage-provisioner/storage-provisioner.template [hunk 2]
```diff
@@ -51,7 +51,7 @@ spec:
   hostNetwork: true
   containers:
   - name: storage-provisioner
-    image: {{default "gcr.io/k8s-minikube" .ImageRepository}}/storage-provisioner-ARCHTAG_NONE:v1.8.1
+    image: {{default "gcr.io/k8s-minikube" .ImageRepository}}/storage-provisioner-{{default "ARCHTAG_NONE" .ArchTag}}:v1.8.1
```
Comment (tstromberg) [inline deploy/addons/storage-provisioner/storage-provisioner.template]: ARCHTAG_NONE feels like a hack, but I don't honestly understand it enough. 

I don't want to violate Chesterton's Fence (https://en.wikipedia.org/wiki/Wikipedia:Chesterton%27s_fence), but I wonder if this template macro just be safely removed with this new model, in preference to `storage-provisioner:v1.1.1`?

I believe all of the mentions of ARCHTAG in these template files should be replaced with {{ .Arch }}, however.

File: Makefile [hunk 3]
```diff
@@ -100,13 +100,6 @@ ifeq ($(GOOS),windows)
...
 
-ifeq ($(GOARCH),amd64)
-	ARCHTAG ?= -amd64
-	ARCHTAG_NONE ?=
-else
-	ARCHTAG ?= -$(GOARCH)
-	ARCHTAG_NONE ?= -$(GOARCH)
-endif
 
 DEPLOYS=\
```
Comment (tstromberg) [inline Makefile]: I believe DEPLOYS should be removable with this change as well. You'll need to rename the .template files back into .yaml though.

**Comment by tstromberg** [comment]
The original pull request by @salamani that added multi-arch support might be useful context:

https://github.com/kubernetes/minikube/pull/4091

**Comment by tstromberg** [comment]
The test failure makes me wonder if this PR needs a rebase, or if I broke head :(

File: deploy/addons/storage-provisioner/storage-provisioner.template (hunk 2 above)
Comment (medyagh) [reply deploy/addons/storage-provisioner/storage-provisioner.template]: fair enough

File: Makefile [hunk 4]
```diff
@@ -229,7 +210,8 @@ out/test.d: pkg/minikube/assets/assets.go
 test:
 	GOPATH=$(GOPATH) ./test.sh
 
-pkg/minikube/assets/assets.go: $(DEPLOYS)
+# Regenerates assets.go when tempalte files have been updated
```
Comment (tstromberg) [inline Makefile]: "template"

**Comment by tstromberg** [comment]
Looks great. Thank you!

**Pull request merged**